ConsoleFileManager 2.0/GUI/Window.cs

```diff
@@ -0,0 +1,170 @@
+            {
+                Console.Write("│");
+                for (int j = X + 1; j < X + Width - 1; j++)
+                    Console.Write(" ");
+                Console.Write("│");
+            }
+            //подвал
+            Console.Write("└");
+            for (int i = 0; i < Width - 2; i++)
+                Console.Write("─");
+            Console.Write("┘");
+            Console.SetCursorPosition(X, Y);
+    }
+
+    public void WriteMesage(string? message, bool newLine)
+    {
+        //if(message != null && message.Length > 0)
+        if (message is { Length: > 0 })
+        {
+            if (newLine)
+            {
+                WriteLine(message);
+            }
+            else
+            {
+                Write(message);
+            }
+        }
+    }
+
+    public double ReadDouble(string? message, bool newLine = true)
+    {
+        bool success;
+        double result;
+        do
+        {
+            WriteMesage(message, newLine);
+
+            var input = Console.ReadLine();
+            success = double.TryParse(input, out result);
+            if (!success)
+                WriteLine("Строка имела неверный формат");
+        }
+        while (!success);
+
+        return result;
+    }
+
+    public int ReadInt(string? message, bool newLine = true)
+    {
+        bool success;
+        int result;
+        do
+        {
+            WriteMesage(message, newLine);
+
+            var input = Console.ReadLine();
+            success = int.TryParse(input, out result);
+            if (!success)
+                WriteLine("Строка имела неверный формат");
+        }
+        while (!success);
+
+        return result;
+    }
+
+    public virtual string ReadLine(string? message, bool newLine = true) { return ""; }
+
+    public virtual void Write(string message)
+    {
+        SetCursorPosition();
+
+        Console.Write(message);
+    }
+
+    public virtual void WriteLine(string message)
+    {
+        SetCursorPosition();
+
+        Console.WriteLine(message);
+    }
+
+    private void SetCursorPosition()
+    {
+        (int left, int top) = GetCursorPosition();
+        if (top > Y + Height - 2)
+        {
+            Console.SetCursorPosition(X, Y + 1);
+            (left, top) = GetCursorPosition();
+        }
+        if (left == 0)
+            Console.SetCursorPosition(left + 1, top);
+    }
+
+    private (int left, int top) GetCursorPosition()
+    {
+        return (Console.CursorLeft, Console.CursorTop);
+    }
+
+    public void Write(StringBuilder tree, string[] args)
+    {
+        int page;
+        int pageLines = Height - 3;
+        string[] lines = tree.ToString().Split('\n');
+        int pageTotal = (lines.Length + pageLines - 1) / pageLines;
+
+        if (args.Length > 2 && args[1] == "-p" && int.TryParse(args[2], out page))
+        {
+            if (page > pageTotal)
+            {
+                page = pageTotal;
+            }
```

Comment:
```suggestion
            page = Math.Min(page, pageTotal);
```
Парадоксально, но так работает быстрее...